decompiler/IR2/FormExpressionAnalysis.cpp

```diff
@@ -4680,9 +4737,13 @@ FormElement* ConditionElement::make_zero_check_generic(const Env& env,
   if (as_logtest) {
     auto logtest_form = pool.alloc_single_form(nullptr, as_logtest);
     auto as_cpad_macro = try_make_logtest_cpad_macro(logtest_form, pool);
+    auto focus_test_macro = try_make_focus_test_macro(logtest_form, pool);
     if (as_cpad_macro) {
       logtest_form = pool.alloc_single_form(nullptr, as_cpad_macro);
     }
+    if (focus_test_macro) {
+      logtest_form = pool.alloc_single_form(nullptr, focus_test_macro);
+    }
```

Comment:
I think you can rewrite this so it doesn't always try to do both macros if one succeeded (they can't both succeed).